index.js

```diff
@@ -0,0 +1,183 @@
+   * @param {String} options.bundledSpecPath - Path to your bundled and processed test suite.
+   * @param {Object} [options.karmaConfig]
+   * @param {Number} [options.karmaConfig.port=9876] - Port on which to run the Karma server.
+   * @param {String} [bundlePath] - Base bath to use for Erik's bundled files. A directory named
+   * `erik` will be created here.
+   */
+  constructor(options) {
+    this._gulp = options.gulp;
+    this._watch = options.watch || false;
+    this._taskDependencies = options.taskDependencies || [];
+    this._localDependencies = options.localDependencies || [];
+    this._remoteDependencies = options.remoteDependencies || [];
+    this._bundledSpecPath = options.bundledSpecPath;
+    this._port = (options.karmaConfig && options.karmaConfig.port) || 9876;
+    this._bundlePath = options.bundlePath || '';
+
+    this._assertValidOptions(options);
+
+    this._erikPath = path.join(options.bundlePath, 'erik');
+
+    this._registerTasks();
+  }
+
+  _assertValidOptions() {
+    if (typeof this._gulp !== 'object') {
+      throw new Error('`options.gulp` is of an invalid type.');
+    }
+
+    if (typeof this._watch !== 'boolean') {
+      throw new Error('`options.watch` is of an invalid type.');
+    }
+
+    if (typeof this._taskDependencies !== 'object') {
+      throw new Error('`options.taskDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._localDependencies !== 'object') {
+      throw new Error('`options.localDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._remoteDependencies !== 'object') {
+      throw new Error('`options.remoteDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._bundledSpecPath !== 'string') {
+      throw new Error('`options.bundledSpecPath` is of an invalid type.');
+    }
+ 
+    if (typeof this._port !== 'number') {
+      throw new Error('`options.karmaConfig.port` is of an invalid type.');
+    }
+ 
+    if (typeof this._bundlePath !== 'string') {
+      throw new Error('`options.bundlePath` is of an invalid type.');
+    }
+   }
+
+  _registerTasks() {
+    this._registerFetchRemoteDeps();
+    this._registerBundleDeps();
+    this._registerCombine();
+    this._registerRunSpec();
+    this._registerErik();
+  }
+
+  _registerFetchRemoteDeps() {
+    this._gulp.task('erik-fetch-remote-deps', () => {
+      return streamqueue({
+        objectMode: true
+      }, ...this._remoteDependencies.map((urlStr) => {
+        const url = URL.parse(urlStr);
+        const base = `${url.protocol}//${url.host}`;
+        const path = url.path;
+
+        return remoteSrc([path], {
+          base
+        });
+      }))
+        .pipe(concat('remote-deps.js'))
+        .pipe(this._gulp.dest(this._erikPath));
+    });
+  }
+
+  _registerBundleDeps() {
+    this._gulp.task('erik-bundle-deps', () => {
+      return this._gulp.src([
+        'erik/remote-deps.js',
```

Comment:
Looks like an empty array here actually caused `streamqueue` to hang and `gulp` to exit. I've added a line to return early if there are no remote dependencies.